xclim/core/indicator.py

```diff
@@ -1309,30 +1331,64 @@ def _postprocess(self, outs, das, params, indexer):
             # We flag periods according to the missing method. skip variables without a time coordinate.
             src_freq = self.src_freq if isinstance(self.src_freq, str) else None
             miss = (
-                self._missing(da, params["freq"], src_freq, options, indexer)
+                self._missing(
+                    da, params["freq"], src_freq, options, params.get("indexer", {})
+                )
                 for da in das.values()
                 if "time" in da.coords
             )
-            mask = reduce(np.logical_or, miss)
+            # Reduce by or and broadcast to ensure the same length in time
+            # When indexing is used and there are no valid points in the last period, mask will not include it
+            mask = reduce(np.logical_or, miss).reindex_like(outs[0], fill_value=True)
```

Comment:
In the case where the last period contains no points after indexing, it doesn't appear in `mask` and so it was dropped from the result, which is a breaking change in a way. 
However, this `reindex_like` trick adds complexity, but fixes the problem.

I would prefer a breaking change than this.

Ex: If you do something with a time series from Jan 1st to Dec 31st, then pass `freq='AS=JUL'` and indexer : `month=2`. The second period, is completely dropped from the mask as it contains no points. Before, it wasn't dropped, rather it was masked out.